### PR TITLE
fix: updated RequestClient to handle proxy from default PROXY_HOST env var

### LINKED ISSUE #336: Feature Request: Add way to specify proxy for requests
### Feature Request

The `request` library we are using offers a way to specify a `proxy` but there is currently no way for developers to pass that into our default `RequestClient`. We should allow to specify it in the configuration of the constructor or find a way to pass it on a per request basis.

**Version:** any

### Code Snippet
```node
// Proposed solution:
// Specify the proxy globally
const client = twilio('accountSid', 'authToken', { proxy: 'proxyUrl' });
```

### PR DESCRIPTION
**Pull request by tlawrie**
Fixes #336 

**Description:**
This fix updates the RequestClient to look at the default PROXY_HOST environment variable. It retains all original options however moves the keep-alive into the Connection header. Removes the http.Agent and https.Agent and replaces with the https-proxy-agent extension to handle the proxy variable. Additionally as mentioned in the Axios issues set the proxy to false (Axios issue reference: https://github.com/axios/axios/issues/2072).

This has been tested in a simple node request. End users do not have to provide any additional information when calling Twilio for a client and as such has no breaking change.

**Test Code:**
Taken from https://www.twilio.com/docs/sms/tutorials/how-to-send-sms-messages-node-js?code-sample=code-send-an-sms-using-the-programmable-sms-api&code-language=Node.js&code-sdk-version=3.x

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the master branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

**Comment by tlawrie** [comment]
Hey @childish-sambino is there anything else required for this?

**Comment by childish-sambino** [comment]
It's on my list of items to review.

File: lib/base/RequestClient.js [hunk 1]
```diff
@@ -41,7 +40,9 @@ RequestClient.prototype.request = function(opts) {
   var deferred = Q.defer();
   var headers = opts.headers || {};
 
-  if (!headers.Connection && !headers.connection) {
+  if (!headers.Connection && !headers.connection && opts.forever) {
+    headers.Connection = 'keep-alive';
```
Comment (childish-sambino) [inline lib/base/RequestClient.js:44]: nit: `headers.Connection = opts.forever ? 'keep-alive' : 'close';`

File: lib/base/RequestClient.js [hunk 2]
```diff
@@ -56,8 +57,8 @@ RequestClient.prototype.request = function(opts) {
     url: opts.uri,
     method: opts.method,
     headers: opts.headers,
-    httpAgent: opts.forever ? new http.Agent({ keepAlive: true }) : undefined,
```
Comment (childish-sambino) [inline lib/base/RequestClient.js:59]: Is `httpAgent` no longer needed?

File: lib/base/RequestClient.js (hunk 1 above)
Comment (tlawrie) [reply lib/base/RequestClient.js:44]: Hi. Apologies. Not following. NIT?

I think you are suggesting there is an alternate better approach here?

File: lib/base/RequestClient.js (hunk 2 above)
Comment (tlawrie) [reply lib/base/RequestClient.js:59]: Tbh. Wasn't quite sure. With my testing of it, and by providing proxy or non proxy, I could use the HTTPSAgent.

File: lib/base/RequestClient.js (hunk 1 above)
Comment (childish-sambino) [reply lib/base/RequestClient.js:44]: [nit](https://stackoverflow.com/questions/27810522/what-does-nit-mean-in-hacker-speak)

Just suggesting a less-verbose replacement that's functionally equivalent.

**Review by childish-sambino — APPROVED** [review]

**Comment by tlawrie** [comment]
Hi. Just checking in if this can be merged yet?

**Comment by tlawrie** [comment]
@thinkingserious 👍

**Pull request merged**